File: App/src/app/Model/clientDTO.ts
export class ClientDTO {
    id!: string; // في الـ C# هو Guid، وفي الـ TypeScript بنعامله كـ string
    name!: string; // Name
    gender!: string; // Gender
    phone!: string; // Phone
    email!: string; // Email
    createdAt?: Date; // CreatedAt (علامة الاستفهام لأنها nullable في الـ C#)
}

File: ServerTransferAPI/ServerTransferAPI/Model/ClientDTO.cs
using AutoMapper;
using System.ComponentModel.DataAnnotations;

namespace ServerTransferAPI.Model
{
    [AutoMap(typeof(Client), ReverseMap = true)]

    public class ClientDTO
    {
        public Guid Id { get; set; } = Guid.NewGuid();


        [Required(ErrorMessage = "الاسم مطلوب")]
        public string Name { get; set; }

        [Required]
        public string Gender { get; set; }

        [Required]
        public string Phone { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        public DateTime? CreatedAt { get; set; }
    }
}